rotkehlchen/db/snapshots.py

```diff
@@ -109,26 +116,53 @@ def export(self, timestamp: Timestamp, directory_path: Optional[Path]) -> Tuple[
             return self.create_zip(
                 timed_balances=timed_balances,
                 timed_location_data=timed_location_data,
+                main_currency=main_currency,
+                timestamp=timestamp,
             )
 
         return self._export(
             timed_balances=timed_balances,
             timed_location_data=timed_location_data,
             directory=directory_path,
+            main_currency=main_currency,
+            timestamp=timestamp,
         )
 
+    @staticmethod
+    def _convert_currency(serialized_data: List[Dict[str, str]], main_currency: Asset, timestamp: Timestamp) -> None:  # noqa: E501
+        if main_currency == A_USD:
+            return
+        column_name = main_currency.symbol.lower() + '_value'
+        for balance in serialized_data:
+            k = PriceHistorian.query_historical_price(A_USD, main_currency, timestamp)
+            balance[column_name] = str(float(balance['usd_value']) * float(k))
+            del balance['usd_value']
```

Comment:
A few things in the implementation of this function. (It's not what I would do -- and I will explain in the next section why -- but first let me say what is wrong)

1. Please don't use variable names like `k`. Always opt for readability such as `price.`
2. Always use kwargs. For example here in `query_historical_price`
3. Never use float. It's not accurate in python and has multiple problems. As you will notice we don't use it anywhere in the codebase. We have our own implementation of decimal called `FVal`. This is what you should use.

-----

Okay so now on the approach. The way you made it, the code serializes the data and then you convert it. But that's wrong. You should convert the data, **before** they are serialized, while they are still in our code's data structures such as `FVal`.

Then the converted data should be serialized and exported. Serialization is always the last step before sending it out to the wire. You should not work on serialized data.

---

Does this make sense?

Reply:
Yeah, thank you very much, this surely makes sense!
The reason why I've chosen this approach is that `DBAssetBalance` is used on a lot of places in the project so I was really afraid to modify it.
So should I modify `DBAssetBalance` structure, create a wrapper on this structure or do smth else?

Reply:
If a NamedTuple needs to be at some point edited I would simply change it to a dataclass that is not frozen and edit it.